File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/sqlite/DomainType.scala
package dev.typr.foundationssc.docs.sqlite
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object DomainType:
  // start
  case class ProductId(value: Long)

  val productIdType: SqliteType[ProductId] =
    SqliteTypes.integer.transform(ProductId.apply, _.value)
  // stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlite/DomainType.java
package dev.typr.foundations.docs.sqlite;

import dev.typr.foundations.SqliteType;
import dev.typr.foundations.SqliteTypes;

@SuppressWarnings("unused")
public class DomainType {
  // start
  // Wrapper type
  public record ProductId(Long value) {}

  // Build a SqliteType from `integer` via a bidirectional transform
  SqliteType<ProductId> productIdType =
      SqliteTypes.integer.transform(ProductId::new, ProductId::value);
  // stop
}
